File: src/main/java/de/tectoast/emolga/commands/draft/AddSDNameCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.sql.DBManagers;

public class AddSDNameCommand extends Command {

    public AddSDNameCommand() {
        super("addsdname", "Registriert deinen Showdown-Namen bei Emolga", CommandCategory.Draft);
        setArgumentTemplate(ArgumentManagerTemplate.builder()
                .add("name", "SD-Name", "Der SD-Name", ArgumentManagerTemplate.Text.any())
                .add("id", "Die ID (nur Flo)", "Nur für Flo", ArgumentManagerTemplate.DiscordType.ID, true)
                .setExample("!addsdname TecToast")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) throws Exception {
        ArgumentManager args = e.getArguments();
        if (args.has("id") && e.isNotFlo()) {
            e.reply("Nur Flo darf den Command mit einer ID verwenden!");
            return;
        }
        String name = args.getText("name");
        boolean b = DBManagers.SD_NAMES.addIfAbsent(name, args.getOrDefault("id", e.getAuthor().getIdLong()));
        if (b) {
            if (args.has("id")) {
                e.getJDA().retrieveUserById(args.getID("id")).queue(mem -> e.reply("Der Name `%s` wurde für %s registriert!".formatted(name, mem.getName())));
            } else {
                e.reply("Der Name `%s` wurde für dich registriert!".formatted(name));
            }
        } else {
            e.reply("Der Name ist bereits vergeben!");
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/draft/AddSDNameCommand.kt
package de.tectoast.emolga.commands.draft

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.database.exposed.SDInsertStatus
import de.tectoast.emolga.database.exposed.SDNamesDB
import de.tectoast.emolga.utils.Constants
import dev.minn.jda.ktx.coroutines.await

object AddSDNameCommand : Command("addsdname", "Registriert deinen Showdown-Namen bei Emolga", CommandCategory.Draft) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("name", "SD-Name", "Der SD-Name", ArgumentManagerTemplate.Text.any())
            .add("id", "Die ID (nur Flo)", "Nur für Flo", ArgumentManagerTemplate.DiscordType.ID, true)
            .setExample("!addsdname TecToast")
            .build()
        slash(false, Constants.G.GILDE)
    }

    override suspend fun process(e: GuildCommandEvent) {
        val args = e.arguments
        val idSpecified = args.has("id")
        if (idSpecified && e.isNotFlo) {
            e.reply("Nur Flo darf den Command mit einer ID verwenden!")
            return
        }
        val name = args.getText("name")
        val id = if (idSpecified) args.getID("id") else e.author.idLong
        val specifiedName = if (idSpecified) e.jda.retrieveUserById(id).await().name else e.author.name
        when (SDNamesDB.addIfAbsent(name, id).await()) {
            SDInsertStatus.SUCCESS -> {
                if (idSpecified) {
                    e.reply("Der Name `$name` wurde für $specifiedName registriert!")
                } else {
                    e.reply("Der Name `$name` wurde für dich registriert!")
                }
            }

            SDInsertStatus.ALREADY_OWNED_BY_YOU -> {
                e.reply("Der Name `$name` gehört bereits ${if (idSpecified) specifiedName else "dir"}!")
            }

            SDInsertStatus.ALREADY_OWNED_BY_OTHER -> {
                e.reply("Der Name ist bereits von jemand anderem belegt! Es wird geprüft, ob dieser Name freigegeben werden kann.")
            }
        }
    }
}
